packages/menu/src/submenu.vue

```diff
@@ -208,6 +208,9 @@
         this.dispatch('ElSubmenu', 'mouse-leave-child');
         clearTimeout(this.timeout);
         this.timeout = setTimeout(() => {
+          if (this.appendToBody) {
+            this.rootMenu.openedMenus = [];
+          }
           !this.mouseInChild && this.rootMenu.closeMenu(this.index);
         }, this.hideTimeout);
```

Comment:
```js
if (this.appendToBody) {
    this.$parent.$el.dispatchEvent(new MouseEvent('mouseleave'));
}
```
More reasonable solution.

Reply:
![menu1](https://user-images.githubusercontent.com/8121621/57667626-d8da2f80-7636-11e9-97a4-8e63ca3e3d43.gif)

This animation is a bit strange.